### ISSUE TITLE
Enhancement: no-namespace - add an option called allowTypeOnlyNamespaces

### ISSUE THREAD

**Issue by k-yle** [typescript-eslint/typescript-eslint#11703]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-namespace/

### Description

Add an option to allow type-only namespaces, but ban namespaces which contain runtime code.

This would align with nodejs's [type-stripping](https://nodejs.org/api/typescript.html#type-stripping), which allows namespaces if they only include type defintions, but [errors](https://nodejs.org/api/errors.html#err_unsupported_typescript_syntax) if a namespace contains runtime code<sup>[[1]](https://nodejs.org/api/typescript.html#typescript-features)</sup>

### Fail

```TypeScript
namespace X {
  export type Y = number;

  const a = 1;
}
```

### Pass

```TypeScript
namespace X {
  export type Y = number;
}
```

### Additional Info

happy to send a PR if this is accepted

**Comment by kirkwaiblinger** [comment]
Hi @k-yle! 

I'd think you'd just want to disable the rule entirely and use TS's [erasableSyntaxOnly option](https://www.typescriptlang.org/tsconfig/#erasableSyntaxOnly)? ([playground](https://typescript-eslint.io/play/#ts=5.9.3&fileType=.tsx&code=FAOwhgtgpgzgDmAxlABAFQJ5ygeRAGwxQG9gUUoAPOAewCcAXFBrVAMRppQF4UByAGac%2BwAL7BQkWAmQoASgFcQDAJbQAclPhJUpclVqMUiGiBhMhXXnwBGYOnwDcYoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUhYAhiVkAjDAGUAnkTyyAHtiKo1FMKPRSAviDNA&tokens=false))

Is there a reason not to go this route?

**Comment by bradzacher** [comment]
As per the rule docs:

> Namespaces are an outdated way to organize TypeScript code. ES2015 module syntax is now preferred

Personally I don't see how this proposal aligns with the rule's intention.

**Comment by k-yle** [comment]
thanks! i didn't realise [`erasableSyntaxOnly`](https://www.typescriptlang.org/tsconfig/#erasableSyntaxOnly) distinguished between runtime namespaces and type-only namespaces. seems like my knowledge is [quite out of date](https://github.com/bloomberg/ts-blank-space/pull/32#issue-2813841349) 😅

I think we can close this then
